test-suite/shared/src/test/scala/org/scalajs/testsuite/niocharset/BaseCharsetTest.scala

```diff
@@ -14,14 +14,13 @@ import scala.util._
 import java.nio._
 import java.nio.charset._
 
-import scala.scalajs.js
-import js.JSConverters._
+import org.junit.Assert._
 
-import org.scalajs.jasminetest.JasmineTest
-
-class BaseCharsetTest(val charset: Charset) extends JasmineTest {
+class BaseCharsetTest(val charset: Charset) {
   import BaseCharsetTest._
 
+  protected def inJVM: Boolean = 1.0.toString != "1"
```

Comment:
Well I guess we could put that test once and for all in `testsuite.Util` or something. But yes, that's our most reliable test. Or we make it a non-shared `object Platform { final val isJVM = false }` in `js/` and `= true` in `jvm/`.